### PR TITLE
Add optimization option `--fcombine-load-threshold=XXX`

### PR DESCRIPTION
**Pull request by nomaddo**
Related #67.

This pullreq add `--fcombine-load-threshold` to adjust the threshold of `CombineLoadingLiterals`.
The impact of the optimization is huge (which is discussed in #67), users may specify the threshold.

It's better to control it by the compiler, but now there is no good way.

The naming rule is as follows:

- optimization options should be long options.
- the prefix of optimization options should be "--f".
- words should be separated by "-". No captalize.

### CODE REVIEW

File: src/optimization/LiteralValues.cpp [hunk 1]
```diff
@@ -791,7 +791,6 @@ InstructionWalker optimizations::handleUseWithImmediate(
                     // and with constant/unsigned, etc.)
                     if(localIt->getSingleWriter() != nullptr)
                         it->addDecorations(intermediate::forwardDecorations(localIt->getSingleWriter()->decoration));
-                    it.nextInBlock();
                     op->replaceLocal(oldLocal, tmp.local, LocalUse::Type::READER);
```
Comment (doe300) [inline src/optimization/LiteralValues.cpp]: Why did you remove this instruction here? Was it a bug?

File: src/main.cpp [hunk 2]
```diff
@@ -101,6 +101,38 @@ static void printInfo()
...
+ */
+Optional<int> parseIntOption(std::string name, std::string input)
+{
+    std::stringstream ss(input);
+    std::string buffer;
+    std::getline(ss, buffer, '=');
+    if(buffer == name)
+    {
+        std::getline(ss, buffer, '=');
+        if(name == buffer)
```
Comment (doe300) [inline src/main.cpp]: Why do you re-test for the name?
I think it would be more obvious what is tested here if you reset buffer to an empty string and here test for the buffer still being empty.

File: src/optimization/LiteralValues.cpp (hunk 1 above)
Comment (nomaddo) [reply src/optimization/LiteralValues.cpp]: This is a bug. Fixed.

File: src/main.cpp (hunk 2 above)
Comment (nomaddo) [reply src/main.cpp]: If `ss` is `--fcombine-load-threshold` (just this, not including '='), `stringstream` reach the end and second `std::getline` do nothing. So `name == buffer` is needed here.
I add comments about this.
